packages/sources/intrinio-test/src/endpoint/price-ws.ts

```diff
@@ -0,0 +1,76 @@
+import { WebSocketTransport } from '@chainlink/external-adapter-framework/transports'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { customSettings } from '../config'
+import { IntrinioRealtime } from './util'
+
+export type IntrinioFeedMessage = {
+  topic: string
+  payload: {
+    type: string
+    timestamp: number
+    ticker: string
+    size: number
+    price: number
+  }
+  event: string
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: { base: string }
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    WsMessage: IntrinioFeedMessage[]
+  }
+}
+
+let ws: IntrinioRealtime
+export const wsTransport = new WebSocketTransport<EndpointTypes>({
+  url: (context) => {
+    const { API_KEY } = context.adapterConfig
+    if (!ws) {
+      ws = new IntrinioRealtime({
+        api_key: API_KEY,
+        provider: 'iex',
+      })
+    }
+    return ws._makeSocketUrl.bind(ws)()
+  },
+  handlers: {
+    open: (connection) => {
+      const heartbeatMsg = JSON.stringify(ws._makeHeartbeatMessage())
+      connection.send(heartbeatMsg)
+      return Promise.resolve()
```

Comment:
This looks a bit funny, could we remove this?

Reply:
I should clarify, I only meant the last `return` line. Does this work without the entire `open()` handler?

Reply:
The `heartBeatMsg` is not necessary, it works the same. But I better leave it, in v2 it was.